backend/appwrite-functions/announcement/ancm-function/main.py

```diff
@@ -0,0 +1,212 @@
+                "status": "success",
+                "sum": len(updated),
+                "updated": updated,
+            }
+        )
+    )
+
+
+def handleRemove(client_payload: dict, database: Database, collection_id: str):
+    anmc_created_dates = client_payload["created_dates"]
+    removed = []
+    for anmc_date in anmc_created_dates:
+        listDocuments = database.list_documents(
+            collection_id=collection_id,  # collection["$id"],
+            order_field="created_at",
+            filters=[f"created_at={anmc_date}"],
+            limit=1,
+        )
+        # Remove the document
+        if len(listDocuments["documents"]) == 1:
+            database.delete_document(
+                collection_id=collection_id,
+                document_id=listDocuments["documents"][0]["$id"],
+            )
+            removed.append(anmc_date)
+    # Success message
+    print(
+        json.dumps(
+            {
+                "action": "removeAnnouncements",
+                "status": "success",
+                "sum": len(removed),
+                "removed": removed,
+            }
+        )
+    )
+
+
+def handleGet(client_payload: dict, database: Database, collection_id: str):
+    # Extract data from payload
+    nbr_ancm = client_payload["numberOfAnnouncements"]
+    timestamp = client_payload["timestamp"]
+    after = client_payload["after"]
+    # Prepare payload to return to client
+    return_payload = {
+        "action": "getAnnouncements",
+        "status": "success",
+        "sum": -1,
+        "announcements": [],
+    }
+    # Query data from collection
+    listDocuments = database.list_documents(
+        collection_id=collection_id,  # collection["$id"],
+        order_field="created_at",
+        order_type="DESC",
+        filters=[f"created_at>={timestamp}"] if after else [f"created_at<={timestamp}"],
+        limit=nbr_ancm,
+    )
+    # Add data to payload
+    return_payload["sum"] = len(listDocuments["documents"])
+    for document in listDocuments["documents"]:
+        return_payload["announcements"].append(
+            {
+                "created_at": document["created_at"],
+                "updated_at": document["updated_at"],
+                "title": document["title"],
+                "content": document["content"],
+            }
+        )
+    # There is no way to return JSON data back directly to client, so we transfer by using stdout.
+    print(json.dumps(return_payload))
+    exit()
+
+
+def main():
+    client = init_client()
+    database = Database(client)
+    PAYLOAD = os.environ.get("APPWRITE_FUNCTION_DATA")
+    if not PAYLOAD:
+        exit()
+    try:
+        client_payload = json.loads(PAYLOAD)
+        if "action" not in client_payload:
+            print_failure_message("'action' key not found in data object!")
+            exit()
+        collection_id = get_collection_id(database=database)
+
+        if client_payload["action"] == "getAnnouncements":
```

Comment:
Is it possible to factor out the function application?
```
handler = handleGet if ... else
    handleAdd if ... else
    handleUpdate if ... else
    handleRemove
```

These awful if-else chains fortunately could be history with the future `match` statement.

Reply:
imo, there is no reason to split that. This main function is as clear as it can get, and move the logic away from main to an extend function will not make things easier to read in practice (more scrolling require).